### PR TITLE
test(alert): typescript refactor

### PR DESCRIPTION
**Pull request by ZhuoyuJin**
### Proposed behaviour

Rename the `alert.cy.js` into `alert.cy.tsx` file in `./cypress/components/alert/` folder
Refactor tests to Typescript
Move any remaining stories in `alert.test.tsx` file to `alert-test.stories.tsx` in './src/components/alert/' folder.

### Current behaviour

tests are in js not ts

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [ ] Related docs have been updated if required

#### QA

- [x] Tested in CodeSandbox/storybook
- [x] Add new Cypress test coverage if required
- [x] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions
- [x] Run `npx cypress open --component` to check if the`alert.cy.tsx` file passed
- [x] Run `npx cypress run --component` to check none of the other *.cy.tsx files have regressed

### CODE REVIEW

File: cypress/components/alert/alert.cy.tsx [hunk 1]
```diff
@@ -110,7 +110,10 @@ context("Testing Alert component", () => {
       [SIZE.EXTRALARGE, 1080],
     ])(
       "should render Alert component with %s as a size and has width property set to %s",
-      (size, width) => {
+      // https://github.com/bahmutov/cypress-each/issues/2
```
Comment (DlgSHi) [inline cypress/components/alert/alert.cy.tsx:113]: no, you need to wrap `[SIZE.EXTRALARGE as string, 1080],` and don't ignore it

**Review by DlgSHi — CHANGES_REQUESTED** [review]

**Review by DlgSHi — DISMISSED** [review]

**Review by divyajindel — APPROVED** [review]

File: cypress/components/alert/alert.cy.tsx [hunk 2]
```diff
@@ -101,13 +101,13 @@ context("Testing Alert component", () => {
...
-      [SIZE.LARGE, 960],
-      [SIZE.EXTRALARGE, 1080],
+      [SIZE.EXTRASMALL as string, 300],
+      [SIZE.SMALL as string, 380],
+      [SIZE.MEDIUMSMALL as string, 540],
+      [SIZE.MEDIUM as string, 750],
+      [SIZE.MEDIUMLARGE as string, 850],
+      [SIZE.LARGE as string, 960],
+      [SIZE.EXTRALARGE as string, 1080],
     ])(
```
Comment (Parsium) [inline cypress/components/alert/alert.cy.tsx:111]: This should help with simplifying the type declaration a little bit, by instead writing the input as an array containing `[string, number]` tuples: 

```tsx
it.each([
      [SIZE.EXTRASMALL, 300],
      [SIZE.SMALL, 380],
      [SIZE.MEDIUMSMALL, 540],
      [SIZE.MEDIUM, 750],
      [SIZE.MEDIUMLARGE, 850],
      [SIZE.LARGE, 960],
      [SIZE.EXTRALARGE, 1080],
    ] as [string, number][])(
```

File: cypress/components/alert/alert.cy.tsx [hunk 3]
```diff
@@ -101,13 +101,13 @@ context("Testing Alert component", () => {
     );
 
     it.each([
```
Comment (Parsium) [inline cypress/components/alert/alert.cy.tsx:103]: This should help with simplifying the type declaration a little bit, by instead writing the input as an array containing `[string, number]` tuples: 

```tsx
it.each([
      [SIZE.EXTRASMALL, 300],
      [SIZE.SMALL, 380],
      [SIZE.MEDIUMSMALL, 540],
      [SIZE.MEDIUM, 750],
      [SIZE.MEDIUMLARGE, 850],
      [SIZE.LARGE, 960],
      [SIZE.EXTRALARGE, 1080],
    ] as [string, number][])(
```

File: cypress/components/alert/alert.cy.tsx [hunk 4]
```diff
@@ -133,7 +133,7 @@ context("Testing Alert component", () => {
       closeIconButton()
         .click()
         .then(() => {
-          // eslint-disable-next-line no-unused-expressions
+          // eslint-disable-next-line no-unused-expressions, jest/valid-expect
```
Comment (Parsium) [inline cypress/components/alert/alert.cy.tsx:136]: We should be ok to disable the jest linting rules here. Ideally, eslint shouldn't be linting the cypress tests with jest rules, so we will need to address this for you as part of FE-5758.

**Review by Parsium — APPROVED** [review]

**Review by Parsium — DISMISSED** [review]

**Comment by divyajindel** [comment]
@ZhuoyuJin  could you please add these below lines top of the alert.cy.tsx file please and remove eslint comment from other places.

/* eslint-disable jest/valid-expect */
/* eslint-disable no-unused-expressions */

**Comment by Parsium** [comment]
> @ZhuoyuJin could you please add these below lines top of the alert.cy.tsx file please and remove eslint comment from other places.
>
> /* eslint-disable jest/valid-expect _/ /_ eslint-disable no-unused-expressions */

I think going forward with the typescript test conversions, disabling the jest eslint rules for the whole file would be good to avoid having to keep disabling line by line - just until we have sorted out the eslint config so the jest rules are not applied to the cypress tests.

However, for non-jest related rules, I would wary of disabling them for the whole test file, since this would mean we are not adhering to the code practices we have set out as a team.

**Review by divyajindel — APPROVED** [review]

File: cypress/components/alert/alert.cy.tsx [hunk 5]
```diff
@@ -108,7 +110,7 @@ context("Testing Alert component", () => {
       [SIZE.MEDIUMLARGE, 850],
       [SIZE.LARGE, 960],
       [SIZE.EXTRALARGE, 1080],
-    ])(
+    ] as [string, number][])(
```
Comment (DlgSHi) [inline cypress/components/alert/alert.cy.tsx:113]: why as `string` and not `AlertProps["size"]`?

**Review by DlgSHi — APPROVED** [review]

**Pull request merged**